lib/github-pages-health-check/domain.rb

```diff
@@ -388,9 +388,19 @@ def enforces_https?
 
       # Can an HTTPS certificate be issued for this domain?
       def https_eligible?
-        (cname_to_github_user_domain? || pointed_to_github_pages_ip?) &&
-          !aaaa_record_present? && !non_github_pages_ip_present? &&
-          caa.lets_encrypt_allowed?
+        # Can't have any IP's which aren't GitHub's present.
+        return false if non_github_pages_ip_present?
+        # Can't have any AAAA records present
+        return false if aaaa_record_present?
+        # Must be a CNAME or point to our IPs.
+
+        # Only check the one domain if a CNAME. Don't check the parent domain.
+        return true if cname_to_github_user_domain?
+
+        # Check CAA records for the full domain and its parent domain.
+        pointed_to_github_pages_ip? &&
+          caa.lets_encrypt_allowed? &&
+          caa.parent_domain_allows_lets_encrypt?
```

Comment:
I don't think this will quite work in this form, as we should only fall back to the parent's `CAA` records if there are no `CAA` records for the custom domain itself.

Reply:
Thinking about this more, now that we are skipping `CAA` completely for the `CNAME` case, we could probably just restore the previous logic, which handled this correctly.